components/unity/unity_port_esp32.c

```diff
@@ -28,6 +29,39 @@ void unity_flush(void)
     esp_rom_uart_tx_wait_idle(CONFIG_ESP_CONSOLE_UART_NUM);
 }
 
+#define iscontrol(c) ((c) <= '\x1f' || (c) == '\x7f')
+
+static void esp_unity_readline(char* dst, size_t len)
+{
+    /* Read line from console with support for echoing and backspaces */
+    size_t write_index = 0;
+    for (;;) {
+        char c = 0;
+        bool got_char = esp_rom_uart_rx_one_char((uint8_t*)&c) == 0;
+        if (!got_char) {
+          continue;
+        }
+        if (c == '\r' || c == '\n') {
+            /* Add null terminator and return on newline */
+            unity_putc('\n');
+            dst[write_index] = '\0';
+            return;
+        } else if (c == '\b') {
+            if (write_index > 0) {
+                /* Delete previously entered character */
+                write_index--;
+                esp_rom_uart_tx_one_char('\b');
+                esp_rom_uart_tx_one_char(' ');
+                esp_rom_uart_tx_one_char('\b');
```

Comment:
Outputting this (`"\b \b"`) when `'\b'` is passed to `unity_putc()`?
If so, I'm not entirely sure if that would be correct as just outputting `\b` alone is valid (it just moves the cursor back one without clearing the character). However, `'\b'` is probably not used currently anyway, so it would probably be fine.

Reply:
Yes, that's what I meant. I thought it would make the function a bit cleaner. LGTM either way. Feel free to leave it as is.